src/Draco.Compiler/Internal/OptimizingIr/Model/IProcedure.cs

```diff
@@ -13,11 +13,22 @@ internal interface IProcedure
     /// </summary>
     public FunctionSymbol Symbol { get; }
 
+    /// <summary>
+    /// Whether this procedure is static or not.
+    /// </summary>
+    public bool IsStatic { get; }
+
     /// <summary>
     /// The name of this procedure.
     /// </summary>
     public string Name { get; }
 
+    /// <summary>
+    /// The type this procedure is defined in.
+    /// When <see langword="null"/>, this procedure is a free function.
+    /// </summary>
+    public IClass? DeclaringType { get; }
```

Comment:
Now procedure has both a declaring type and module, maybe we want to merge them into a declaring entity/container? Because this will severely complicate usage of these members, you'll have to keep checking for both for codegen for example.

Reply:
I suggest to merge theses properties when they get actually used, I don't know yet which common information they want to use right now.